collect_app/src/test/java/org/odk/collect/android/utilities/SQLiteUtilsTest.java

```diff
@@ -0,0 +1,74 @@
+import org.odk.collect.android.application.Collect;
+import org.robolectric.RobolectricTestRunner;
+
+import static android.content.Context.MODE_PRIVATE;
+import static android.provider.BaseColumns._ID;
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.odk.collect.android.database.helpers.FormsDatabaseHelper.FORMS_TABLE_NAME;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.AUTO_DELETE;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.AUTO_SEND;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.BASE64_RSA_PUBLIC_KEY;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.DATE;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.DESCRIPTION;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.DISPLAY_NAME;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.FORM_FILE_PATH;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.FORM_MEDIA_PATH;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.JRCACHE_FILE_PATH;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.JR_FORM_ID;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.JR_VERSION;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.LANGUAGE;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.LAST_DETECTED_FORM_VERSION_HASH;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.MD5_HASH;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.SUBMISSION_URI;
+
+@RunWith(RobolectricTestRunner.class)
+public class SQLiteUtilsTest {
+
+    @Test
+    public void doesColumnExistTest() {
+        SQLiteDatabase db = Collect.getInstance().openOrCreateDatabase("TestDatabase", MODE_PRIVATE, null);
+        db.execSQL("CREATE TABLE IF NOT EXISTS " + FORMS_TABLE_NAME + " ("
+                + _ID + " integer primary key, "
+                + DISPLAY_NAME + " text not null, "
+                + DESCRIPTION + " text, "
+                + JR_FORM_ID + " text not null, "
+                + JR_VERSION + " text, "
+                + MD5_HASH + " text not null, "
+                + DATE + " integer not null, " // milliseconds
+                + FORM_MEDIA_PATH + " text not null, "
+                + FORM_FILE_PATH + " text not null, "
+                + LANGUAGE + " text, "
+                + SUBMISSION_URI + " text, "
+                + BASE64_RSA_PUBLIC_KEY + " text, "
+                + JRCACHE_FILE_PATH + " text not null, "
+                + AUTO_SEND + " text, "
+                + AUTO_DELETE + " text, "
+                + LAST_DETECTED_FORM_VERSION_HASH + " text);");
+
+        assertTrue(SQLiteUtils.doesColumnExist(db, FORMS_TABLE_NAME, _ID));
```

Comment:
I really think you only need one of these assertions and that the database doesn't need to be related to Collect at all! But this is already much easier to understand and much faster to run so I'm fine with leaving it and simplifying in the future if needed.

Reply:
Yeah I think to make it really clear that the behaviour of `doesColumnExist` is independent from the structure of Collect's database it could just create a database with `COLUMN_A` and `COLUMN_B`, test they exist and then test that `COLUMN_C` doesn't. This means you don't have to import the contants from `FormsProviderAPI`. It would also cut down on the size of the test potentially making it easier for someone new to the codebase to understand quickly.

Reply:
ok done.

Reply:
Love it!